From 310ba76ba26e61d59a4f59b30dd3bd9d28add8ca Mon Sep 17 00:00:00 2001
From: Sidhanta Sahu <quic_sidhanta@quicinc.com>
Date: Sat, 21 Oct 2023 11:39:37 -0700
Subject: [PATCH] wifi: mac80211: Fix association with AP_VLAN interface

Post Mode 0 Subsystem Recovery (SSR), DVLAN grouped stations are not
getting associated with MLO AP interface. AP sends AUTH response with
MLD mac address as destination address when AP fails to obtain TX_STA
context using sta_info_get with AP_VLAN interface. And STA expects to
receive AUTH response with LINK mac address.

Fix to obtain TX_STA context from sta_info_get_bss makes AP to send
AUTH response with LINK mac address.

Signed-off-by: Sidhanta Sahu <quic_sidhanta@quicinc.com>
---
 net/mac80211/tx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -1287,7 +1287,7 @@ ieee80211_tx_prepare(struct ieee80211_su
 			tx->sta = sta_info_get_bss(sdata, hdr->addr1);
 		}
 		if (!tx->sta && !is_multicast_ether_addr(hdr->addr1)) {
-			tx->sta = sta_info_get(sdata, hdr->addr1);
+			tx->sta = sta_info_get_bss(sdata, hdr->addr1);
 			aggr_check = true;
 		}
 	}
